cmd/notation/verify.go

```diff
@@ -97,17 +107,15 @@ func runVerify(command *cobra.Command, opts *verifyOpts) error {
 		// TODO: need to change MaxSignatureAttempts as a user input flag or
 		// a field in config.json
 		MaxSignatureAttempts: math.MaxInt64,
+		UserMetadata: userMetadata,
 	}
 
 	// core verify process
 	_, outcomes, err := notation.Verify(ctx, verifier, sigRepo, verifyOpts)
 	// write out on failure
 	if err != nil || len(outcomes) == 0 {
 		if err != nil {
-			var errorVerificationFailed *notation.ErrorVerificationFailed
-			if !errors.As(err, &errorVerificationFailed) {
-				return fmt.Errorf("signature verification failed: %w", err)
-			}
+			return fmt.Errorf("signature verification failed: %w", err)
```

Comment:
whats the implication of this change?

Reply:
if there's any error coming from notation.Verify (and not just ErrorVerificationFailed), the message is returned to the user. looking at this again, there's a bug since if `len(outcomes) == 0`, err can't be nil, so this can be simplified. overall, the implication is that error messages returned from notation.Verify will be displayed to users.

Reply:
> if there's any error coming from notation.Verify (and not just ErrorVerificationFailed), the message is returned to the user. looking at this again, there's a bug since if `len(outcomes) == 0`, err can't be nil, so this can be simplified. overall, the implication is that error messages returned from notation.Verify will be displayed to users.

Looks like this is changing the original output behavior.
The original output design:
1. `err != nil`
       `err is NOT notation.ErrorVerificationFailed: "signature verification failed: err"`
       `err is notation.ErrorVerificationFailed: "signature verification failed for all the signatures associated with ref"`
2. `len(outcomes) == 0` (IMO, we shouldn't rely on `if len(outcomes) == 0, err can't be nil` as this is a logic from the underneath library. If the output pattern in notation-go changed for some reason, then our assumption here will no longer be true.)
       `"signature verification failed for all the signatures associated with ref"`

After this change:
1. `err != nil`
        `"signature verification failed: err"`
2. `len(outcomes) == 0`
        `"signature verification failed for all the signatures associated with ref"`

Note, if `err != nil AND err is notation.ErrorVerificationFailed`, this means that we have gone through all the signatures associated with the artifact and all the signatures failed verification. In this case, we do not need to print out the `err` itself, as the error msg is just `"signature verification failed"`.

Reply:
ah, ok then, can match the original output design then